internal/cli/capact/components_test.go

```diff
@@ -0,0 +1,124 @@
+package capact
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"testing"
+	"time"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	"github.com/stretchr/testify/assert"
+	"helm.sh/helm/v3/pkg/action"
+	"helm.sh/helm/v3/pkg/kube"
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/cli-runtime/pkg/resource"
+)
+
+func TestCreateObjectRetrySuccess(t *testing.T) {
+	tests := map[string]struct {
+		givenError error
+	}{
+		"Should pass on nil error": {
+			givenError: nil,
+		},
+		"Should ignore AlreadyExist error": {
+			givenError: apierrors.NewAlreadyExists(schema.GroupResource{}, "test-object"),
+		},
+	}
+	for tn, tc := range tests {
+		t.Run(tn, func(t *testing.T) {
+			failingCli := &FailingKubeClient{
+				CreateError: tc.givenError,
+			}
+
+			actionCfg := &action.Configuration{
+				KubeClient: failingCli,
+			}
+
+			objToCreate := fmt.Sprintf(issuerTemplate, clusterIssuerName, certManagerSecretName)
+
+			// when
+			err := createObject(actionCfg, []byte(objToCreate))
+
+			// then
+			assert.NoError(t, err)
+			assert.Equal(t, 1, failingCli.CreateCallsCnt)
+		})
+	}
+}
+
+func TestCreateObjectRetryFailed(t *testing.T) {
+	if testing.Short() {
+		t.Skip("Skipping TestCreateObjectRetry as it takes up to 2s")
+	}
+
+	// given
+	expTotalRetryDurationInSec := 1.6
+
+	internalError := apierrors.NewInternalError(errors.New("simulate internal error"))
+	failingCli := &FailingKubeClient{
+		CreateError: internalError,
+	}
+
+	actionCfg := &action.Configuration{
+		KubeClient: failingCli,
+	}
+
+	objToCreate := fmt.Sprintf(issuerTemplate, clusterIssuerName, certManagerSecretName)
+
+	// when
+	startedCall := time.Now()
+	err := createObject(actionCfg, []byte(objToCreate))
+
+	// then
+	assert.EqualError(t, err, internalError.Error())
+	assert.Equal(t, 5, failingCli.CreateCallsCnt)
+	assert.InDelta(t, expTotalRetryDurationInSec, time.Since(startedCall).Seconds(), 0.2)
```

Comment:
Not sure if we should care that much about the time duration. Personally, I would remove this assertion. Another thing is that even if we don't do actual calls, I think there might be some rare VM slowdowns which could potentially cause this test to fail.